src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1436,13 +1436,25 @@ public void changed(Database.ChangeEvent event) {
 
             RevisionInternal rev = change.getAddedRevision();
 
+            // https://github.com/couchbase/couchbase-lite-android/issues/683
+            // DocumentChange of Database.ChangeEvent might not contain revision body.
+            // In case of include_docs=true, needs to load revision body
+            if (changesIncludesDocs && (rev.getBody() == null || !rev.getBody().hasContent())) {
+                EnumSet<TDContentOptions> options = getContentOptions();
+                try {
+                    rev = db.loadRevisionBody(rev, options);
+                } catch (CouchbaseLiteException e) {
+                    Log.w(Log.TAG_ROUTER, "Error loading revision body", e);
+                }
+            }
+
```

Comment:
The code here is out-of-sync with the iOS implementation. There are details about includesConflicts & includesDocs and the winning revision that should be ported here.

iOS Implementation: https://github.com/couchbase/couchbase-lite-ios/blob/master/Source/CBLRouter%2BChanges.m#L182-L201

[API docs] (
http://developer.couchbase.com/documentation/mobile/1.1.0/develop/references/couchbase-lite/rest-api/database/get-changes/index.html) for more info about includeConflicts and includesDocs.